src/Symfony/Bridge/Doctrine/Middleware/Debug/Middleware.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Middleware\Debug;
+
+use Doctrine\DBAL\Driver as DriverInterface;
+use Doctrine\DBAL\Driver\Middleware as MiddlewareInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * Middleware to collect debug data.
+ *
+ * @author Laurent VOULLEMIER <laurent.voullemier@gmail.com>
+ */
+class Middleware implements MiddlewareInterface
```

Comment:
```suggestion
final class Middleware implements MiddlewareInterface
```

There's no value in extending this class. Also, we're implementing an interface from a third-party package. Finalizing the class could make adapting to upstream changes easier in the long run.

I'd also finalize `Driver`, `Connection` and `Statement` for the same reasons.

Reply:
Classes changed to final